DesignspaceEditor2.roboFontExt/lib/designspaceEditor/tools.py

```diff
@@ -196,3 +197,51 @@ def getObjectValue_forString_errorDescription_(self, value, string, error):
         except Exception:
             pass
         return False, string, error
+    
+    
+def postScriptNameTransformer(familyName, styleName):
+    def filterPSName(name):
+        # Define the set of forbidden characters
+        forbidden_chars = set('-[](){}<>/%% ')
+
+        # Normalize the Unicode string to NFKD form
+        name = unicodedata.normalize('NFKD', name)
+
+        # Ensure the name is encoded as ASCII
+        name = name.encode("ascii", errors="ignore").decode()
+
+        # Filter out forbidden characters
+        filtered_name = ''.join(c for c in name if 33 <= ord(c) <= 126 and c not in forbidden_chars)
+
+        return filtered_name
+    
+    # TODO: spec says max length is 63, but should we check this
```

Comment:
@typemytype yeah but just doing it crudely could result in something like `AVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongNameHere-` which since it ends with a hyphen and doesn't include a style name might cause some application somewhere to complain... though it is unlikely...

I made the function a little more smart, it'll take from the family name up to a point then start taking from the style name until it reaches the limit... I guess it doesn't cover every situation but 63 characters is actually pretty long!

Reply:
super smart!! thanks